autometa/validation/datasets.py

```diff
@@ -0,0 +1,174 @@
+Shaurya Chanana, Izaak Miller, Jason C. Kwan
+
+This file is part of Autometa.
+
+Autometa is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Autometa is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with Autometa. If not, see <http://www.gnu.org/licenses/>.
+COPYRIGHT
+
+pulling data from google drive folder with simulated or synthetic communities
+"""
+
+
+import gdown
+import os
+import logging
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+# add function to get the file ids from the user input
+def fetchIDs(folder: str, file: str):
+    """
+    Parameters
+    ----------
+    folder : str
+        specifies the folder that the user would like to download from, if any
+    file : str
+        specifies the file that the user would like to download, if any
+
+    Returns
+    -------
+    dict
+        nested dictionary with the user's specified files' folder, file, and file_id
+        format: {'folder': {<index>: <folder>}, 'file': {<index>: <file>}, 'file_id': {<index>: <file_id>}}
+
+    """
+    # create the dataframe here
+    index = pd.read_csv("gdown_fileIDs.csv", dtype=str)
+
+    # retrieve only the entries that the user wants
+    if folder == None:
+        target_folder = index
+    else:
+        target_folder = index.query(f'folder == "{folder}"')
+
+    if file == None:
+        target_files = target_folder
+    else:
+        target_files = target_folder.query(f'file == "{file}"')
+
+    targets_dict = target_files.to_dict()
+    return targets_dict
+
+
+def download(targets_dict, out_dirpath: str) -> None:
+
+    """Downloads the files specified in a dictionary.
+
+    Parameters
+    ----------
+    targets_dict : dict
+        nested dictionary with the user's specified files' folder, file, and file_id
+        format: {'folder': {<index>: <folder>}, 'file': {<index>: <file>}, 'file_id': {<index>: <file_id>}}
+    out_dirpath : str
+        directory path where the user wants to download the file(s)
+
+    Returns
+    -------
+    None
+        download is completed through gdown
+
+    """
+
+    key_list = [*targets_dict["file"]]
+    for key in key_list:
+        # retrieve values from targets_dict
+        file = targets_dict["file"][key]
+        folder = targets_dict["folder"][key]
+        file_id = targets_dict["file_id"][key]
+
+        # construct file id into a url to put into gdown
+        url = f"https://drive.google.com/uc?id={file_id}"
+        filename = f"{folder}_{file}"
```

Comment:
I'm a little confused by this line. What is the intended behavior?

It appears the directory's and file's names are being joined together into one string rather than placing the file in the provided `folder` directory. I think the most intuitive arguments for the `download` function would be

```python
def download(dataset:str, file: str, out:str)->None:
```

This will require two look-ups:

1. Look for `dataset` of available datasets (could be a datasets dict)
2. Look for `file` in dataset (could be a dict corresponding to respective dataset)

The corresponding `file_id` should be retrieved after successful lookup of 1 and 2.

Now with the `file_id` you can continue as you have done by constructing the url and performing the gdown download. In this case the gdown call uses the provided `out` variable.

e.g.

```python
gdown.download(url, out)
```

By the way, does `gdown.download(...)` return anything? A response code perhaps? If so, could you also return this and handle appropriately? One example would be if an error response is returned, this could be propagated back to the end-user.

Reply:
I don't think gdown returns anything.